bot/exts/fun/anagram.py

```diff
@@ -0,0 +1,102 @@
+import asyncio
+import json
+import logging
+import random
+from pathlib import Path
+
+import discord
+from discord.ext import commands
+
+from bot.bot import Bot
+from bot.constants import Colours, NEGATIVE_REPLIES
+
+log = logging.getLogger(__name__)
+
+TIME_LIMIT = 60
+
+# anagram.json file contains all the anagrams
+with open(Path("bot/resources/fun/anagram.json")) as f:
+    ANAGRAMS_ALL = json.loads(f.read_text("utf8"))
+
+
+class Anagram(commands.Cog):
+    """Cog for the Anagram game command."""
+
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.winners = set()
+        self.correct = ""
+        self.current_channel = None
+
+    @commands.command(name="anagram", aliases=("anag", "gram", "ag"))
+    async def anagram_command(self, ctx: commands.Context) -> None:
+        """
+        Given shuffled letters which can be rearranged to form Anagrams.
+
+        It has code for anagram game command. Users can play the game by using ".anagram" command,
+        it will show an embed with scrambled letters which if rearranged can form words.
+        It selects random choices from a json file which has all the anagrams possible from popular English words.
+        After 60 seconds if anyone has provided a correct answer in channel,
+        it will list out the winner's name and if nobody gets its correct
+        then all the correct answers will be displayed after a minute.
+        """
+        if self.current_channel:
+            await ctx.send(f"An anagram is already being solved in {self.current_channel.mention}!")
+            return
+
+        # Don't let users start in a DM
+        if not ctx.guild:
```

Comment:
Instead of doing this, just add the commands.guild_only() decorator under commands.command

Reply:
Added the commands.guild_only() decorator